xgbgen/main.go

```diff
@@ -12,6 +12,10 @@ import (
 var (
 	protoPath = flag.String("proto-path",
 		"/usr/share/xcb", "path to directory of X protocol XML files")
+	packageName = flag.String("package",
+		"xkb", "name of package to generate in the format (package)/(package).go")
```

Comment:
Oh, that’s how you’re controlling the output name. OK. I get it now.

In that case, I would probably flip things then, and make the default `""`, then if `packageName != "" && packageName != "-"` (pretty standard POSIX pattern), it would then `Create` the file. Then we only need the one flag, and we don’t automatically break any prior usage of the tool. (It would be pretty unexpected that when I forget to pass in `-package foo` it would just happily and silently write it out to `xkb/xkb.go`)

It might also help if the flag were `output-package` as well. I hadn’t initially put the 2 and 2 together that it was controlling the output, in part because these things tend to take something like “package” and use that as the name of the package during generation, and not necessarily control the output itself.

Reply:
Not sure why I didnt realize that putting xkb as the value meant it was the default value... I do agree that this way is better, fixed with https://github.com/jezek/xgb/pull/27/commits/eadbcad00879d7b1aeedca39f23885d8da6486ec